pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -126,19 +127,14 @@ private SegmentCompletionFSM createFsm(LLCSegmentName llcSegmentName, String msg
     SegmentZKMetadata segmentMetadata = _segmentManager.getSegmentZKMetadata(realtimeTableName, segmentName, null);
     Preconditions.checkState(segmentMetadata != null, "Failed to find ZK metadata for segment: %s", segmentName);
     SegmentCompletionFSM fsm;
-    if (segmentMetadata.getStatus() == CommonConstants.Segment.Realtime.Status.DONE) {
-      // Segment is already committed
-      StreamPartitionMsgOffsetFactory factory = getStreamPartitionMsgOffsetFactory(llcSegmentName);
-      StreamPartitionMsgOffset endOffset = factory.create(segmentMetadata.getEndOffset());
-      fsm = SegmentCompletionFSM.fsmInCommit(_segmentManager, this, llcSegmentName, segmentMetadata.getNumReplicas(),
-          endOffset);
-    } else if (msgType.equals(SegmentCompletionProtocol.MSG_TYPE_STOPPED_CONSUMING)) {
-      fsm = SegmentCompletionFSM.fsmStoppedConsuming(_segmentManager, this, llcSegmentName,
-          segmentMetadata.getNumReplicas());
-    } else {
-      // Segment is in the process of completing, and this is the first one to respond. Create fsm
-      fsm = SegmentCompletionFSM.fsmInHolding(_segmentManager, this, llcSegmentName, segmentMetadata.getNumReplicas());
-    }
+
+    String factoryName = _segmentCompletionConfig.getDefaultFsmScheme();
```

Comment:
Should we take scheme from table streaming config, and use default only if it is not configured?

Reply:
good idea, now taking this from stream config map